Classes/Package.php

```diff
@@ -1,74 +1,24 @@
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
 use Neos\Eel\FlowQuery\FlowQuery;
 use Neos\Flow\Core\Bootstrap;
-use Neos\ContentRepository\Domain\Model\Node;
 use Neos\Flow\Package\Package as BasePackage;
 
 /**
  * The Package class, wiring signal/slot during boot.
  */
 class Package extends BasePackage
 {
-    private const NODE_TYPE_IDENTIFIER_MIXIN = 'Neos.Form.Builder:IdentifierMixin';
 
     /**
      * @param Bootstrap $bootstrap The current bootstrap
      * @return void
      */
     public function boot(Bootstrap $bootstrap)
     {
-        $dispatcher = $bootstrap->getSignalSlotDispatcher();
-
-        $dispatcher->connect(Node::class, 'nodePropertyChanged', function (NodeInterface $node, $propertyName, $_, $newValue) {
-            if ($propertyName !== 'identifier' || empty($newValue) || !$node->getNodeType()->isOfType(self::NODE_TYPE_IDENTIFIER_MIXIN)) {
-                return;
-            }
-
-            $this->setUniqueFormElementIdentifier($node, $newValue);
-        });
-
-        $dispatcher->connect(Node::class, 'nodeAdded', function (NodeInterface $node) {
-            try {
-                $identifier = $node->getProperty('identifier');
-
-                if (empty($identifier) || !$node->getNodeType()->isOfType(self::NODE_TYPE_IDENTIFIER_MIXIN)) {
-                    return;
-                }
-            } catch (\Neos\ContentRepository\Exception\NodeException $e) {
-                return;
-            }
-
-            $this->setUniqueFormElementIdentifier($node, $identifier);
-        });
-    }
-
-    /**
-     * @param NodeInterface $node
-     * @param string $identifier
-     * @throws \Neos\Eel\Exception
-     */
-    private function setUniqueFormElementIdentifier(NodeInterface $node, string $identifier): void
-    {
-        /** @noinspection PhpUndefinedMethodInspection */
-        $flowQuery = (new FlowQuery([$node]))->context([
-            'invisibleContentShown' => true,
-            'removedContentShown' => true,
-            'inaccessibleContentShown' => true
-        ]);
-        $possibleIdentifier = $identifier;
-        $i = 1;
-        /** @noinspection PhpUndefinedMethodInspection */
-        while ($flowQuery
-                ->closest('[instanceof Neos.Form.Builder:NodeBasedForm]')
-                // [identifier=".."] matches the Form Element identifier, [_identiier!="..."] excludes the current node
-                ->find(sprintf('[instanceof %s][identifier="%s"][_identifier!="%s"]',
-                    self::NODE_TYPE_IDENTIFIER_MIXIN ,$possibleIdentifier, $node->getIdentifier()))
-                ->count() > 0) {
-            $possibleIdentifier = $identifier . '-' . $i++;
-        }
-        $node->setProperty('identifier', $possibleIdentifier);
+        # BREAKING in Neos 9: No node signals anymore
+        # Missing here: Setting of identifier for Neos.Form.Builder:NodeBasedForm on nodePropertyChanged and nodeAdded
```

Comment:
@bwaidelich Any better idea then a CatchUpHook here? Or are we fine with setting this manual for now?

Reply:
That's a good question.
Currently there is no easy way to achieve the same, but it should be fairly easy™ to add one.. I'll investigate and create an issue.
For the time being I think it's ok to live without the signal

Reply:
This will be solvable with https://github.com/neos/neos-development-collection/issues/5335